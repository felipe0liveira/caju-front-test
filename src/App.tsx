import Router from "~/router";
import { Header } from "./components/Header";
import { QueryClientProvider, QueryClient } from "react-query";

export const queryClient = new QueryClient();

function App() {
  return (
    <>
      <Header>
        <h1>Caju Front Teste</h1>
      </Header>
      <QueryClientProvider client={queryClient}>
        <Router />
      </QueryClientProvider>
    </>
  );
}

export default App;
