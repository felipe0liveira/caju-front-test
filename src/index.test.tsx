import App from "./App";
import { render } from "@testing-library/react";
import { describe, test, expect } from "@jest/globals";

describe("<App />", () => {
  test("Should match snapshot", () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  });
});
