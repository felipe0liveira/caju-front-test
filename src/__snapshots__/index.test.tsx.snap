// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-feUYzb bBIQqf"
      >
        <h1>
          Caju Front Teste
        </h1>
      </header>
      <div
        style="margin-top: 64px;"
      >
        <div
          class="sc-kAycRU iFYosF"
        >
          <div
            class="sc-imWZod tygxB"
          >
            <svg
              aria-hidden="true"
              class="spin"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-aYaIB fRMDGN"
        >
          <div
            class="sc-iGgVNO ddLifx"
          >
            <div>
              <label />
              <input
                class="sc-dcJtft cOQjOK"
                placeholder="Digite um CPF válido"
                value=""
              />
              <span
                style="font-size: 12px; color: red;"
              />
            </div>
            <div
              class="sc-gsFSjX ga-dmrm"
            >
              <button
                aria-label="refetch"
                class="sc-fqkwJk jVmdFk"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
              <button
                class="sc-gEvDqW lgAOsV"
              >
                Nova Admissão
              </button>
            </div>
          </div>
          <div
            class="sc-jXbVAB gFbubx"
          >
            <div
              class="sc-dhKdPU flWiik"
              status="REVIEW"
            >
              <h3
                class="sc-kpDprT jcDgBW"
                status="REVIEW"
              >
                Pronto para revisar
              </h3>
              <div
                class="sc-dAlxHm RzmJa"
              />
            </div>
            <div
              class="sc-dhKdPU jrIju"
              status="APPROVED"
            >
              <h3
                class="sc-kpDprT cBjDKS"
                status="APPROVED"
              >
                Aprovado
              </h3>
              <div
                class="sc-dAlxHm RzmJa"
              />
            </div>
            <div
              class="sc-dhKdPU duxfLR"
              status="REPROVED"
            >
              <h3
                class="sc-kpDprT blgyNC"
                status="REPROVED"
              >
                Reprovado
              </h3>
              <div
                class="sc-dAlxHm RzmJa"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-feUYzb bBIQqf"
    >
      <h1>
        Caju Front Teste
      </h1>
    </header>
    <div
      style="margin-top: 64px;"
    >
      <div
        class="sc-kAycRU iFYosF"
      >
        <div
          class="sc-imWZod tygxB"
        >
          <svg
            aria-hidden="true"
            class="spin"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-aYaIB fRMDGN"
      >
        <div
          class="sc-iGgVNO ddLifx"
        >
          <div>
            <label />
            <input
              class="sc-dcJtft cOQjOK"
              placeholder="Digite um CPF válido"
              value=""
            />
            <span
              style="font-size: 12px; color: red;"
            />
          </div>
          <div
            class="sc-gsFSjX ga-dmrm"
          >
            <button
              aria-label="refetch"
              class="sc-fqkwJk jVmdFk"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              class="sc-gEvDqW lgAOsV"
            >
              Nova Admissão
            </button>
          </div>
        </div>
        <div
          class="sc-jXbVAB gFbubx"
        >
          <div
            class="sc-dhKdPU flWiik"
            status="REVIEW"
          >
            <h3
              class="sc-kpDprT jcDgBW"
              status="REVIEW"
            >
              Pronto para revisar
            </h3>
            <div
              class="sc-dAlxHm RzmJa"
            />
          </div>
          <div
            class="sc-dhKdPU jrIju"
            status="APPROVED"
          >
            <h3
              class="sc-kpDprT cBjDKS"
              status="APPROVED"
            >
              Aprovado
            </h3>
            <div
              class="sc-dAlxHm RzmJa"
            />
          </div>
          <div
            class="sc-dhKdPU duxfLR"
            status="REPROVED"
          >
            <h3
              class="sc-kpDprT blgyNC"
              status="REPROVED"
            >
              Reprovado
            </h3>
            <div
              class="sc-dAlxHm RzmJa"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
